### PR TITLE
Change internal popover markup

### PR DESCRIPTION
**Pull request by oliverviljamaa**
# v4.1.2
## Change internal popover markup

Previously, popover created a container element for the trigger passed as a child and the popover itself, and overrode styles to make the popover itself not be styled like the container that acted as a trigger.

Now, popover decorates the trigger passed as a child with accessibility props and an event handler, removing the need for a wrapper element and custom styles.

This also enables focus style support out of the box.

### CODE REVIEW

File: src/popover/Popover.js [hunk 1]
```diff
@@ -47,27 +46,30 @@ export default class Popover extends Component {
...
   };
 
+  createTrigger = () => {
+    const { children: child } = this.props;
+
+    return cloneElement(typeof child === 'string' ? <span>{child}</span> : child, {
+      'data-toggle': 'popover',
+      role: 'button',
+      tabIndex: 0,
+      onClick: this.open,
```
Comment (KarlEST) [inline src/popover/Popover.js]: no more `onKeyPress`?

File: src/popover/Popover.js [hunk 2]
```diff
@@ -47,27 +46,30 @@ export default class Popover extends Component {
     document.removeEventListener('click', this.closePopoverOnOutsideClick, true);
   };
 
+  createTrigger = () => {
+    const { children: child } = this.props;
+
+    return cloneElement(typeof child === 'string' ? <span>{child}</span> : child, {
```
Comment (KarlEST) [inline src/popover/Popover.js]: Not sure I understand correctly but doesn't `string` one need the `onClick` etc as well to open the popover?

Comment (oliverviljamaa) [reply src/popover/Popover.js]: The second argument, the options object, is not part of the ternary, and is added whatever the first argument is.

File: src/popover/Popover.js (hunk 1 above)
Comment (oliverviljamaa) [reply src/popover/Popover.js]: No, `tabIndex` ensures it's keyboard navigatable.

File: src/popover/Popover.js (hunk 2 above)
Comment (Tankenstein) [reply src/popover/Popover.js]: Curious: Why do you need to put it in a span if it's a string?

Comment (oliverviljamaa) [reply src/popover/Popover.js]: Sometimes the child is not a React element, but a string. In that case, I need a wrapper element for attaching keypress and click listeners.

Comment (Tankenstein) [reply src/popover/Popover.js]: right, cloneelement does not work for strings 🤦‍♂️

**Pull request merged**